cmd/test.go

```diff
@@ -0,0 +1,38 @@
+package cmd
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type testCase struct {
+	name      string
+	input     []string
+	stdoutput string
+	erroutput string
+	err       error
+}
+
+func executeTestCases(t *testing.T, testCases []testCase) {
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			stdBuf := new(bytes.Buffer)
+			errBuf := new(bytes.Buffer)
+
+			rootCmd.SetOut(stdBuf)
+			rootCmd.SetErr(errBuf)
+			rootCmd.SetArgs(tc.input)
+
+			err := rootCmd.Execute()
+			if tc.err == nil {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, tc.err, err.Error())
```

Comment:
This will panic in the case of expecting an error, but getting none :) In that case `err` will be `nil` and `err.Error()` will panic.

The `assert.EqualError()` function takes the actual err as second argument so it should be

```go
assert.EqualError(t, err, tc.err.Error())
```